ta/pkcs11/src/pkcs11_attributes.c

```diff
@@ -938,11 +1001,29 @@ enum pkcs11_rc check_created_attrs_against_processing(uint32_t proc_id,
 	case PKCS11_PROCESSING_IMPORT:
 		assert(check_attr_bval(proc_id, head, PKCS11_CKA_LOCAL, false));
 		break;
+	case PKCS11_CKM_GENERIC_SECRET_KEY_GEN:
+	case PKCS11_CKM_AES_KEY_GEN:
+		assert(check_attr_bval(proc_id, head, PKCS11_CKA_LOCAL, true));
+		break;
 	default:
 		TEE_Panic(proc_id);
 		break;
 	}
 
+	switch (proc_id) {
+	case PKCS11_CKM_GENERIC_SECRET_KEY_GEN:
+		if (get_key_type(head) != PKCS11_CKK_GENERIC_SECRET)
```

Comment:
could be asserted since implementation checks/mandates these.

Reply:
Agreed. We are already checking for this in create_attributes_from_template() and returning error from there. So, here we can just add an assert to check of this mismatch is still there. I will make the change.